### PR TITLE
Allow delete all questions

### LINKED ISSUE #12159: EQM: When deleting "quizzes" some do not get deleted
## Observed behavior
when deleting  quizzes at once, some don't get deleted

## Errors a

https://github.com/learningequality/kolibri/assets/103313919/c95645fd-31a9-4b8a-802a-7b2be218dd91

## Expected behavior
When a user select all and click delete everything  should be deleted

## Steps to reproduce
go to add questions > select quizzes and click apply changes.
click select all > delete

## Context

Develop

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Summary

After discussion with @rtibbles, we propose this solution to #12159. Where we have the possibility of eliminating all questions from a section. And to avoid possible unintended consequences, add a validation to disable the save quiz button if all sections are empty.

## References

Closes #12159.

## Reviewer guidance

Create an empty quiz, add sections without adding resources, and try to save it.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/composables/useQuizCreation.js [hunk 1]
```diff
@@ -509,6 +507,7 @@ export default function useQuizCreation() {
   provide('activeQuestions', activeQuestions);
   provide('selectedActiveQuestions', selectedActiveQuestions);
   provide('allQuestionsSelected', allQuestionsSelected);
+  provide('allSectionsEmpty', allSectionsEmpty);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/composables/useQuizCreation.js]: I don't think this needs to be provided/injected, as we are only ever consuming it in the place that `useQuizCreation` is invoked.

**Review by nucleogenesis — COMMENTED** [review]
One question before approval:

If we save empty sections, do we then display them to the Learner or should we just remove empty sections upon save? It might be worth even just notifying the user in that case like "Sections A, B and C have no questions. If you start the quiz for the learners, all empty sections will be removed."

---

Otherwise, the code overall looks good, but I thought of one place where we could make an optimization:

I don't think this is blocking as it'd require more manual testing and could be deferred to a follow-up issue.

Here in [updateSection](https://github.com/learningequality/kolibri/blob/16cf3c8dc3ca48db0b76f5a3a39ebef7ad94c8f6/kolibri/plugins/coach/assets/src/composables/useQuizCreation.js#L104-L126) - I think we can skip all of this if `question_count` is being set to 0.

As if there aren't already enough nested if statements here, one final one like "if(question_count.value === 0) { updates.question_count = question_count; updates.questions = []; } else { ... the other code ... }"

Basically this will save the user's device a few cycles by not iterating unnecessarily over the original resource pool.

**Comment by AlexVelezLl** [comment]
Hi @nucleogenesis. Yes, Richard suggested to remove them on publish, I opened an issue for this: https://github.com/learningequality/kolibri/issues/12217.

> Here in [updateSection](https://github.com/learningequality/kolibri/blob/16cf3c8dc3ca48db0b76f5a3a39ebef7ad94c8f6/kolibri/plugins/coach/assets/src/composables/useQuizCreation.js#L104-L126) - I think we can skip all of this if question_count is being set to 0.

I will take a look!

**Comment by AlexVelezLl** [comment]
Thanks @nucleogenesis! I have pushed the changes :open_hands:

**Review by nucleogenesis — APPROVED** [review]
